patches/api/0501-Extend-HumanEntity-dropItem-API.patch

```diff
@@ -0,0 +1,108 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Strokkur24 <strokkur24@gmail.com>
+Date: Sat, 30 Nov 2024 11:54:38 +0100
+Subject: [PATCH] Extend HumanEntity#dropItem API
+
+
+diff --git a/src/main/java/org/bukkit/entity/HumanEntity.java b/src/main/java/org/bukkit/entity/HumanEntity.java
+index 488604ba1a516b477693877c74712e4a45624a8b..2aab711ece14f008f2c4bd1b46d4e48cc621b547 100644
+--- a/src/main/java/org/bukkit/entity/HumanEntity.java
++++ b/src/main/java/org/bukkit/entity/HumanEntity.java
+@@ -706,6 +707,97 @@ public interface HumanEntity extends LivingEntity, AnimalTamer, InventoryHolder
+      */
+     public boolean dropItem(boolean dropAll);
+ 
++    // Paper start - Extend HumanEntity#dropItem API
++    /**
++     * Makes the entity drop an item from their inventory.
++     * <br>
++     * This method calls {@link HumanEntity#dropItem(int slot)}
++     * with the first {@link ItemStack} occurrence in the inventory
++     *
++     * @param itemStack The ItemStack to drop
++     * @return The dropped item, or null if the action was unsuccessful
++     */
++    public Item dropItem(final ItemStack itemStack);
++
++    /**
++     * Makes the entity drop an item from their inventory.
++     * <br>
++     * This method calls {@link HumanEntity#dropItem(int slot, boolean retainOwnership)}
++     * with the first {@link ItemStack} occurrence in the inventory
++     *
++     * @param itemStack The ItemStack to drop from their inventory
++     * @param retainOwnership Whether this player should own the resulting item entity
```

Comment:
retainOwnership is very misleading (from nms itself) it is not setting the owner but the thrower
the difference here is that if an item is owned it can only be picked up by this very owner
the thrower is just the entity's uuid that threw the item
potentially a less misleading name would be appropriate or clarification in the docs

Reply:
Ah okay, thanks for clarifying that. It mislead even me 😅